resources/views/pages/user/[user]/game/[game]/compare.blade.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+use function Laravel\Folio\{middleware, name};
+
+middleware(['auth', 'can:view,game', 'can:view,user']);
+name('game.compare-unlocks');
+
+?>
+
+@props([
+    'user' => null,
+    'otherUser' => null,
+    'game' => null,
+    'achievements' => [],
+    'sortOrder' => 'display',
+])
+
+@php
+use App\Enums\Permissions;
+
+$availableSorts = [
+    'selfUnlocks' => 'My Unlock Times',
+    'otherUnlocks' => 'Their Unlock Times',
+    'display' => 'Display Order',
+    'title' => 'Achievement Title',
+];
+
+$userUnlockCount = 0;
+$otherUserUnlockCount = 0;
+$userUnlockHardcoreCount = 0;
+$otherUserUnlockHardcoreCount = 0;
+$numAchievements = count($achievements);
+
+
+foreach ($achievements as $achievement) {
+    if (array_key_exists('userTimestamp', $achievement)) {
+        $userUnlockCount++;
+        if ($achievement['userHardcore'] ?? false) {
+            $userUnlockHardcoreCount++;
+        }
+    }
+
+    if (array_key_exists('otherUserTimestamp', $achievement)) {
+        $otherUserUnlockCount++;
+        if ($achievement['otherUserHardcore'] ?? false) {
+            $otherUserUnlockHardcoreCount++;
+        }
+    }
+}
+
+$canModerate = ($user->Permissions >= Permissions::Moderator);
+
+@endphp
+
+<x-app-layout
+    pageTitle="Compare Unlocks - {{ $game->Title }}"
+    pageDescription="Compares unlocks between {{ $user->User }} and {{ $otherUser->User }} for {{ $game->Title }}"
+>
+    <x-user.breadcrumbs
+        :targetUsername="$otherUser->User"
+        :parentPage="$game->Title"
+        :parentPageUrl="$game->permalink"
+        currentPage="Compare Unlocks"
+    />
+
+    <div class="mt-3 mb-3 w-full relative flex gap-x-3">
+        {!! gameAvatar($game->toArray(), label: false, iconSize: 48, iconClass: 'rounded-sm') !!}
+        <h1 class="mt-[10px] w-full">Compare Unlocks</h1>
+
+        @if ($canModerate)
+            <x-hidden-controls-toggle-button>Moderate</x-hidden-controls-toggle-button>
+        @endif
+    </div>
+
+@if ($canModerate)
+    <x-hidden-controls>
+        <div style="width:20%">
+            <ul class="flex flex-col gap-2">
+                <x-game.link-buttons.game-link-button
+                    icon="🔬"
+                    href="/usergameactivity.php?ID={{ $game->ID }}&f={{ $otherUser->User }}"
+                >
+                    View User Game Activity
+                </x-game.link-buttons.game-link-button>
+            </ul>
+        </div>
+    </x-hidden-controls>
+@endif
+
+@if (@$numAchievements === 0)
```

Comment:
Is the `@` directly before `$numAchievements` a typo? Maybe just `@if (!$numAchievements)` would suffice.

Reply:
`@if` simply performs a truthiness evaluation of the given expression. The expression `!$numAchievements` is close to functionally equivalent given that 0 is a falsy value (though this expression also checks for stuff like `null`). 

However, the difference between the two is usually pedantic and explicitly checking for `$numAchievements === 0` is also perfectly ok.